app/code/Magento/Backend/Model/Menu/Config.php

```diff
@@ -63,6 +63,11 @@ class Config
      */
     protected $_appState;
 
+    /**
+     * @var Builder
+     */
+    protected $_menuBuilder;
```

Comment:
Please add `_menuBuilder` property with the `private` scope of visibility.

Reply:
@dmytro-ch I understand as per Magento's guidelines every field needs to be made private, however every-other field on the class seems to been marked protected, hence i marked this too as protected. Let me know if you still believe this needs to be marked as private.

Reply:
@sankalpshekhar all properties by default should be private. The protected properties provide a possibility to extend the class but the composition should be used instead of inheritance. These protected properties are legacy fields.

Reply:
@joni-jones I understnad that, but refer to @sidolov comment down below.

> By default dynamically declared filed has public visibility modifier, that means that property accessible from in the all child classes, declaring such properties as private we will eliminate this behavior. I'll suggest you declare mentioned properties at least with protected visibility modifier to avoid breaking changes.

Reply:
Looks reasonable for me to keep it protected. I'm resolving this conversation